source/Irrlicht/CBawFile.h

```diff
@@ -107,6 +144,15 @@ namespace irr { namespace core
 		uint64_t attrBufPtrs[scene::EVAI_COUNT];
 		uint64_t idxBufPtr;
 	} PACK_STRUCT;
+
+	struct FinalBoneHierarchyBlobV1 : Blob
```

Comment:
see my note about variable size blobs

Reply:
About variable size blob: I'm allocating memory on stack for it (see for example CBAWMeshWriter.cpp:54), it's your idea actually. So now `VariableSizeBlob` would allocate mem on heap for us... At most it could make sense for `CFinalBoneHierarchy` and corresponding blob struct right now

Reply:
Well if we know the blob to be less than 16kb then lets placement new it on the stack, but for things which can be much bigger like the CFinalBoneHierarchy (as it contains all the animations), we definitely need the heap.

So that VariableSizeBlob::caclBlobSizeForObj static function could help us differentiate between the two cases.

We dont have to call "allocMemForBlob"

It was more of an idea to have a static create function which requires you to say how much memory you reserved or allocated and assert if the constructor would overrun it, and that needs some sort of mechanism to allow us to create the object via a placement new but having the constructor declared as private.